### DISCUSSION TITLE
ebusctl options -f and -m not working with crontab

### DISCUSSION THREAD

**Discussion by maxx37** [john30/ebusd#1225] [Q&A]
Hello all,

I have a weird problem: when I  do

      /usr/bin/ebusctl read -f -c bai Flame

in terminal window, it works fine, with or without the full path to ebusctl.

If I put the same thing in a cronjob, it does't work anymore, with or without full path.

This holds also when this line is used in a script file that is called by cron.

After trying a lot, I found out that the problem is specifically linked to the -f option, and also to the -m option.
Other options like -v or -n are working perfectly, both in cron , script files called by cron, and on command line, even with several options combined.

Apparently, some kind of process that  does the forced read from the bus is not starting from within cron.

Anybody has an idea for a workaround ?

Beste regards, Marc

**Comment by jstoki** [comment]
Hi
I have a similar problem with cron. It looks like when running via cron, the process did not wait for a response from ebusctl. Something like a timeout. This is visible with the “-m” option. Sometimes even two commands in a row for the same parameter, the first one will respond and the second one will not.

**Cron:**
`*/2` * * * * /home/stoko/script/status_ebus.sh >> /home/stoko/script/status_ebus.log 2>&1

### Script:
```
date '+%Y%m%d_%H%M%S'
echo "Channel:Ogrzewanie Field:TempZasobnikGora ebusctl read -m 59 -n -c vr_71 SensorData1 s5"
ebusctl read -m 59 -n -c vr_71 SensorData1 s5
echo "Channel:Ogrzewanie Field:TempZasobnikDol ebusctl read -m 59 -n -c vr_71 SensorData1 s6"
ebusctl read -m 59 -n -c vr_71 SensorData1 s6
echo "Channel:Ogrzewanie,CieplaWoda Field:TempZewnetrzna ebusctl read -m 59 -n -c bai Status01 temp_2"
ebusctl read -m 59 -n -c bai Status01 temp_2
echo "Channel:CieplaWoda, Field:TempSolarDach ebusctl read -m 59 -n -c vr_71 SensorData1 s7"
ebusctl read -m 59 -n -c vr_71 SensorData1 s7
echo "Channel:CieplaWoda, Field:Cyrkulacja ebusctl read -m 59  -n -c bai Cirpump"
ebusctl read -m 59  -n -c bai Cirpump
echo "Channel:Ogrzewanie, Field:TempPietro ebusctl read -m 59  -n -c 700 z1RoomTemp"
ebusctl read -m 59  -n -c 700 z1RoomTemp
echo "Channel:Ogrzewanie q1, Field:TempParter ebusctl read -m 59  -n -c 700 z2RoomTemp"
ebusctl read -m 59  -n -c 700 z1RoomTemp
echo "Channel:Ogrzewanie q2, Field:TempParter ebusctl read -m 59  -n -c 700 z2RoomTemp"
ebusctl read -m 59  -n -c 700 z2RoomTemp
echo "Channel:Ogrzewanie, Field:PompaPodloga ebusctl read -m 59  -n -c 700 Hc2PumpStatus"
ebusctl read -m 59  -n -c 700 Hc2PumpStatus
echo "Channel:Ogrzewanie, Field:PompaKaloryfer ebusctl read -m 59  -n -c 700 Hc1PumpStatus"
ebusctl read -m 59  -n -c 700 Hc1PumpStatus
echo "Channel:Ogrzewanie, Field:TempZasilaniaPodloga ebusctl read -m 59  -n -c 700 Hc2FlowTemp"
ebusctl read -m 59  -n -c 700 Hc2FlowTemp
echo "Channel:Ogrzewanie, Field:TempZasilaniaGrzejnik ebusctl read -m 59  -n -c 700 Hc1FlowTemp"
ebusctl read -m 59  -n -c 700 Hc1FlowTemp
echo "Palnik ebusctl read -m 59 -n -c bai Flame"
ebusctl read -m 59 -n -c bai Flame
```
**Log:**
**20260118_103401**
Channel:Ogrzewanie Field:TempZasobnikGora ebusctl read -m 59 -n -c vr_71 SensorData1 s5
42.62
Channel:Ogrzewanie Field:TempZasobnikDol ebusctl read -m 59 -n -c vr_71 SensorData1 s6
10.62
Channel:Ogrzewanie,CieplaWoda Field:TempZewnetrzna ebusctl read -m 59 -n -c bai Status01 temp_2
-8.375
Channel:CieplaWoda, Field:TempSolarDach ebusctl read -m 59 -n -c vr_71 SensorData1 s7
-5.38
Channel:CieplaWoda, Field:Cyrkulacja ebusctl read -m 59  -n -c bai Cirpump
0
Channel:Ogrzewanie, Field:TempPietro ebusctl read -m 59  -n -c 700 z1RoomTemp
22.125
Channel:Ogrzewanie q1, Field:TempParter ebusctl read -m 59  -n -c 700 z2RoomTemp
22.125
Channel:Ogrzewanie q2, Field:TempParter ebusctl read -m 59  -n -c 700 z2RoomTemp
21.3125
Channel:Ogrzewanie, Field:PompaPodloga ebusctl read -m 59  -n -c 700 Hc2PumpStatus
1
Channel:Ogrzewanie, Field:PompaKaloryfer ebusctl read -m 59  -n -c 700 Hc1PumpStatus
1
Channel:Ogrzewanie, Field:TempZasilaniaPodloga ebusctl read -m 59  -n -c 700 Hc2FlowTemp
28.3125
Channel:Ogrzewanie, Field:TempZasilaniaGrzejnik ebusctl read -m 59  -n -c 700 Hc1FlowTemp
33.375
Palnik ebusctl read -m 59 -n -c bai Flame
15
**20260118_103601**
Channel:Ogrzewanie Field:TempZasobnikGora ebusctl read -m 59 -n -c vr_71 SensorData1 s5
42.56
Channel:Ogrzewanie Field:TempZasobnikDol ebusctl read -m 59 -n -c vr_71 SensorData1 s6
10.62
Channel:Ogrzewanie,CieplaWoda Field:TempZewnetrzna ebusctl read -m 59 -n -c bai Status01 temp_2
-8.375
Channel:CieplaWoda, Field:TempSolarDach ebusctl read -m 59 -n -c vr_71 SensorData1 s7
-5.31
Channel:CieplaWoda, Field:Cyrkulacja ebusctl read -m 59  -n -c bai Cirpump
0
**Channel:Ogrzewanie, Field:TempPietro ebusctl read -m 59  -n -c 700 z1RoomTemp
Channel:Ogrzewanie q1, Field:TempParter ebusctl read -m 59  -n -c 700 z2RoomTemp
22.125**
**Channel:Ogrzewanie q2, Field:TempParter ebusctl read -m 59  -n -c 700 z2RoomTemp**
Channel:Ogrzewanie, Field:PompaPodloga ebusctl read -m 59  -n -c 700 Hc2PumpStatus
1
**Channel:Ogrzewanie, Field:PompaKaloryfer ebusctl read -m 59  -n -c 700 Hc1PumpStatus
Channel:Ogrzewanie, Field:TempZasilaniaPodloga ebusctl read -m 59  -n -c 700 Hc2FlowTemp
Channel:Ogrzewanie, Field:TempZasilaniaGrzejnik ebusctl read -m 59  -n -c 700 Hc1FlowTemp
Palnik ebusctl read -m 59 -n -c bai Flame**
